services/director/src/simcore_service_director/producer.py

```diff
@@ -997,6 +1008,26 @@ async def get_service_details(app: web.Application, node_uuid: str) -> Dict:
             ) from err
 
 
+@retry(
+    wait=wait_fixed(2),
+    stop=stop_after_delay(10),
```

Comment:
Here you are assuming the reply arrives very fast (which should be the case). If you have an error while the stop is in progress, this will not work, since this call usually takes a very long time.
I would move this to a counter (stop after 3 times).

Reply:
very good point